src/lib.rs

```diff
@@ -1073,31 +1089,63 @@ impl AssociatedType {
     }
 }
 
+// ===== impl Field =====
+
+impl Field {
+    /// Return a field definition with the provided name and type
+    pub fn new<T>(name: &str, ty: T) -> Self
+    where T: Into<Type>,
+    {
+        Field {
+            name: name.into(),
+            ty: ty.into(),
+            documentation: String::new(),
+            annotation: String::new(),
+        }
+    }
+
+    /// Set field's documentation.
+    pub fn with_documentation(&mut self, documentation: &str) -> &mut Self {
+        self.documentation = documentation.into();
+        self
+    }
+
+    /// Set field's annotation.
+    pub fn with_annotation(&mut self, annotation: &str) -> &mut Self {
+        self.annotation = annotation.into();
+        self
+    }
+}
+
 // ===== impl Fields =====
 
 impl Fields {
-    fn named<T>(&mut self, name: &str, ty: T) -> &mut Self
-    where T: Into<Type>,
+    fn push_named(&mut self, field: Field) -> &mut Self
     {
         match *self {
             Fields::Empty => {
-                *self = Fields::Named(vec![Field {
-                    name: name.to_string(),
-                    ty: ty.into(),
-                }]);
+                *self = Fields::Named(vec![field]);
             }
             Fields::Named(ref mut fields) => {
-                fields.push(Field {
-                    name: name.to_string(),
-                    ty: ty.into(),
-                });
+                fields.push(field);
             }
             _ => panic!("field list is named"),
-        }
+        };
```

Comment:
I don't think that this semicolon is needed.

Reply:
Hum! interesting. I though the match would return... Fixed in 95b6673.